### PR TITLE
Fix gpgpu example so that the glsl actually affects all elements

### PR DESCRIPTION
**Pull request by avl**
Previously the glsl code only acted on every 4th element.

A comment: This is how I understood from tomaka that it should be. I do think I may have misunderstood you, tomaka! Please give me feedback how you would like it to work :-)

### CODE REVIEW

File: examples/gpgpu.rs [hunk 1]
```diff
@@ -6,28 +6,34 @@ use glium::glutin;
...
-                float values[256];
+                vec4 values[4096/4];
             };
 
             void main() {
-                float val = values[gl_GlobalInvocationID.x];
-                values[gl_GlobalInvocationID.x] = pow(val, power);
+                vec4 val = values[gl_GlobalInvocationID.x];
+                
+                values[gl_GlobalInvocationID.x] = vec4(
```
Comment (tomaka) [inline examples/gpgpu.rs]: No need to change this line. `pow(val, power);` should work as well.

Comment (avl) [reply examples/gpgpu.rs]: I didn't think of that! However, now that I tried it it seems it doesn't actually work. The glsl pow() function is only defined for scalars. At least my nvidia 970 drivers on linux don't accept a vec4 as first parameter.

Comment (avl) [reply examples/gpgpu.rs]: Or maybe both arguments need to be vec4. Didn't try that. Sorry for speaking too quickly!

**Comment by avl** [comment]
I rewrote the pow call so there is only one call for all 4 elements.

**Comment by avl** [comment]
I also rebased my branch ontop of glium master.

File: examples/gpgpu.rs [hunk 2]
```diff
@@ -14,20 +14,21 @@ fn main() {
...
-                float values[256];
+                vec4 values[4096/4];
             };
 
             void main() {
-                float val = values[gl_GlobalInvocationID.x];
-                values[gl_GlobalInvocationID.x] = pow(val, power);
+                vec4 val = values[gl_GlobalInvocationID.x];
+
+                values[gl_GlobalInvocationID.x] = pow(val, vec4(power,power,power,power));
```
Comment (tomaka) [inline examples/gpgpu.rs]: Should be `vec4(power)` instead of `vec4(power, power, power, power)`.

**Pull request merged**